Documentation for Federated (OIDC) Keystone adoption

xek opened the pull request:
- **Documentation for Federated (OIDC) Keystone adoption**

Jira: https://issues.redhat.com/browse/OSPRH-19963
Jira: https://issues.redhat.com/browse/OSPRH-19960

xek committed:
Documentation for Federated (OIDC) Keystone adoption

xek commented:
Reopening this PR to resurrect the federation adoption documentation work.

**Status:** The branch is currently 234 commits behind main and needs a rebase before it can be merged.

@xek Could you please rebase this branch on main to bring it up to date?

```bash
git fetch upstream
git rebase upstream/main
git push --force-with-lease
```

Once rebased, this should be ready for review. Note: This documentation PR complements PR #1061 (federation adoption tests).

klgill commented on docs_user/modules/proc_configuring-federation-for-keystone.adoc:
```diff
@@ -0,0 +1,117 @@
++
+* Replace `keycloak-ca.crt` with the path to the CA file you want to use.
+
+. Create the `keystone-httpd-override` secret that provides the Apache HTTPD overrides required for OIDC:
```
```suggestion
. Create the `keystone-httpd-override` secret that provides the Apache HTTPD overrides that are required for OIDC:
```

klgill commented on docs_user/modules/proc_configuring-federation-for-keystone.adoc:
```diff
@@ -0,0 +1,117 @@
+    </LocationMatch>
+----
++
+* Replace the placeholder values, such as `<client_secret>` and `<crypto_passphrase>`, with the credentials that match your Keycloak deployment.
```
```suggestion
+
where:

`<client_secret>`:: Replace with your client ID to use for the OIDC provider handshake. You must get this from your SSO administrator.
`<crypto_passphrase>`:: Replace with the client secret to use for the OIDC provider handshake. You must get this from your SSO administrator after providing your redirect URLs.
`<LocationMatch>` and `<Location>`:: Replace with the chosen string that creates your unique redirect URL.
```
@xek I got these descriptions from the following guide. Please confirm that they are relevant for this step.
https://docs.redhat.com/en/documentation/red_hat_openstack_services_on_openshift/18.0/html/configuring_security_services/assembly_rhoso-federation#proc_deploying-rhoso-with-a-federated-idp_rhoso-federation

klgill reviewed (changes requested):
@xek Please let me know if you have questions about any of my edits. Thanks!

xek committed:
Address review comments from klgill

- Fix _mod-docs-content-type attribute syntax (add colon prefix)
- Use double quotes for id attribute
- Change "To let" to "To allow" for stronger verb
- Use angle brackets for placeholder values in commands
- Replace bullet-point replacements with "where:" definition lists
- Change "required for" to "that are required for"
- Add detailed descriptions for client_secret, crypto_passphrase placeholders
- Fix "access token issued" to "access token that is issued"
- Fix "<project id>" to "<project_id>" with where clause

afaranha commented on docs_user/modules/proc_configuring-federation-for-keystone.adoc:
```diff
@@ -0,0 +1,117 @@
+        customConfigSecret: keystone-httpd-override
+----
++
+* Ensure you merge the patch with any existing custom configuration for the {identity_service}.
```
Yes, this patch is to update their keystone configuration, it'll just add this new configuration for OIDC.

afaranha commented on docs_user/modules/proc_configuring-federation-for-keystone.adoc:
```diff
@@ -0,0 +1,117 @@
+    OS_ACCESS_TOKEN=$(openstack token issue -f value -c id) \
+    openstack project show <project id>
+----
++
```
That's right, this is to test the new token, the user should use a project they have access to for it to work.

afaranha commented on docs_user/modules/proc_configuring-federation-for-keystone.adoc:
```diff
@@ -0,0 +1,117 @@
+    openstack project show <project id>
+----
++
+A successful response confirms that the federated OIDC configuration is active on the podified Keystone deployment.
```
It depends on the user's environment, in this example, the response should be the details of the project, it'll change for each project.

afaranha reviewed (commented):
@klgill just replied to your queries.
Also opened a new PR since this one was closed https://github.com/openstack-k8s-operators/data-plane-adoption/pull/1397
